src/mdyml/convert_cisagov.py

```diff
@@ -160,15 +162,31 @@ def convert() -> None:
 
         out_dict_list.append(out_dict)
 
-    doc = {"version": "1.0", "software": out_dict_list}
-
-    yaml = ruamel.yaml.YAML()
-    yaml.indent(mapping=2, offset=2, sequence=4)
-    yaml.explicit_start = True
-    yaml.explicit_end = True
-    yaml.sort_base_mapping_type_on_output = False
-    yaml.allow_unicode = True
-    yaml.dump(doc, sys.stdout)
+    out_dict_groups = {
+        k: list(g)
+        for k, g in groupby(out_dict_list, key=lambda s: s["vendor"][0].upper())
+    }
+
+    non_letter_groups = list()
+    for key in list(out_dict_groups.keys()):
+        if not key.isalpha():
+            non_letter_groups.extend(out_dict_groups[key])
+            del out_dict_groups[key]
+    out_dict_groups["0-9"] = non_letter_groups
```

Comment:
Are there any funky cases where the leading character of a vendor name is not alphanumeric?  It might be better to name this group `out_dict_groups["nonalpha"]` since you are really only checking that.

Reply:
Nothing in our data contained that, but yes that is a case we should consider. I was mainly mirroring how the data is split up in https://github.com/NCSC-NL/log4shell/tree/main/software honestly.

Reply:
Given that this is international and supports Unicode, there could be some oddballs in there.

This is a good time to remind @felddy to meditate upon The Solidus.

Reply:
Speaking of Unicode we should _probably_ be normalizing the mapping keys...

Reply:
I agree with @jsf9k and prefer `"nonalpha"` to `"0-9"` here.

Reply:
Please see d42b79b for your consideration.